### PR TITLE
Google Sheets: Trimmed zero-width unicode characters

### PR DESCRIPTION
**Pull request by madclaws**
## Summary
 Target issue is #4165  

Trimming zero-width unicode characters from gsheet keys and values. Since these were causing problem when we fetch by key from DB.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved handling of sheet data by automatically removing invisible Unicode characters from keys and values, ensuring cleaner and more consistent data processing.

- **Tests**
  - Added a test to verify that sheet data containing invisible Unicode characters is correctly cleaned and normalized during sheet creation.

### CODE REVIEW

File: lib/glific/third_party/sheets/sheets.ex [hunk 1]
```diff
@@ -19,6 +19,8 @@ defmodule Glific.Sheets do
     Sheets.SheetData
   }
 
+  # zero width unicode characters
+  @invisible_unicode_range ~r/[\x{200B}-\x{200F}\x{202A}-\x{202E}\x{2060}-\x{206F}\x{FEFF}]/u
```
Comment (shijithkjayan) [inline lib/glific/third_party/sheets/sheets.ex:23]: **question:** Is this all the invible unicode characters? Can you please share the link to the source, if you have one?

Comment (madclaws) [reply lib/glific/third_party/sheets/sheets.ex:23]: These are from a pov of [typesetting](https://ptiglobal.com/the-beauty-of-unicode-zero-width-characters/) which causes these kind of issues when users copy-paste from say a pdf. Excluding this there are a lot more, but here we are focusing on the important  typeset chars

File: lib/glific/third_party/sheets/sheets.ex [hunk 2]
```diff
@@ -407,4 +414,11 @@ defmodule Glific.Sheets do
 
     :ok
   end
+
+  @spec trim_string(String.t()) :: String.t()
+  defp trim_string(str) do
+    str
+    |> String.trim()
```
Comment (shijithkjayan) [inline lib/glific/third_party/sheets/sheets.ex:421]: **question:** Should we also add a test that checks we trim the key and values and not just replace the invisible unicode characters?

**Review by AmishaBisht — APPROVED** [review]

**Review by akanshaaa19 — APPROVED** [review]

File: lib/glific/third_party/sheets/sheets.ex (hunk 2 above)
Comment (madclaws) [reply lib/glific/third_party/sheets/sheets.ex:421]: Added one. When i tested with google sheets, it was automatically trimming there itself even if added spaces..

File: lib/glific/third_party/sheets/sheets.ex [hunk 3]
```diff
@@ -192,7 +194,7 @@ defmodule Glific.Sheets do
 
           %{
             ## we can also think in case we need first column.
-            key: row["Key"],
+            key: parsed_rows.values["key"],
```
Comment (shijithkjayan) [inline lib/glific/third_party/sheets/sheets.ex:197]: **question:** Why this change?

File: lib/glific/third_party/sheets/sheets.ex (hunk 2 above)
Comment (shijithkjayan) [reply lib/glific/third_party/sheets/sheets.ex:421]: In that case, do we even need this here?

Comment (madclaws) [reply lib/glific/third_party/sheets/sheets.ex:421]: its better us handling it, since google sheet is a 3rd party and changes are not in our control.

File: lib/glific/third_party/sheets/sheets.ex (hunk 3 above)
Comment (madclaws) [reply lib/glific/third_party/sheets/sheets.ex:197]: Because `parsed_rows.values["key"]` will be going through all the parsing which includes trimming.
